core/src/main/scala/org/broadinstitute/dsde/rawls/spendreporting/SpendReportingService.scala

```diff
@@ -509,20 +517,50 @@ class SpendReportingService(
         )
       }
 
-      query = getQuery(aggregations, spendExportConf)
-      queryJob = setUpQuery(query, spendExportConf, start, end, projectNames)
+      // look for a cached spend report
+      isCacheable = aggregationsAreCacheable(aggregations)
+      cachedResults <-
+        if (isCacheable)
+          getCachedSpendReportData(projectNames, start, end)
+        else
+          Future.successful(CachedSpendReportingResults(isCacheValid = false, results = None))
+
+      spendResults <-
+        if (cachedResults.isCacheValid) {
+          rawlsCacheHitRate().hit()
+          cachedResults.results match {
+            case Some(report) => Future.successful(report)
+            case None =>
+              throw RawlsExceptionWithErrorReport(
+                StatusCodes.NotFound,
+                s"no spend data found for billing project ${project.value} between dates ${toISODateString(start)} and ${toISODateString(end)}"
+              )
+          }
+        } else {
+          rawlsCacheHitRate().miss()
```

Comment:
this records a cache miss even when `isCacheable` is false

Reply:
hmmm … @samanehsan you're driving the metrics; what do you think here? When a user requests a per-billing-project spend report and asks for a daily view - which the cache doesn't support - should this log a cache miss or should it not contribute to hit/miss metrics at all?

Reply:
Hmm since it's not supported by the cache I think it makes sense to leave it out